larg4/Services/LArG4Detector_service.cc

```diff
@@ -321,24 +321,34 @@ void larg4::LArG4DetectorService::setStepLimits(G4GDMLParser *parser){
                   << " appropriate parameter.";
 
   std::string volumeName  = "";
-  float previousStepLimit = 0.;
-  float newStepLimit      = 0.;
-  for(size_t i=0; i<inputVolumes_; ++i)
+  G4LogicalVolume* setVol = 0;
+  for(auto const& [name, newStepLimit] : overrideGDMLStepLimit_Map )
   {
+    G4double previousStepLimit = 0.;
+
     // -- Check whether the volumeName provided corresponds to a valid volumeName in the geometry
-    G4LogicalVolume* setVol = parser->GetVolume(selectedVolumes_[i].first);
+    try {
+      setVol = volStore->GetVolume(name);
+    } catch (...)
+    {
+      throw cet::exception("invalidInputVolumeName") << "Provided volume name : "
+                                                     << name << " not found!\n";
+      continue;
+    }
```

Comment:
Got it, thanks! 

I also removed the input argument to that function since we can get the VolumeStore instance like in your example above -- no need to pass it the reference from the `doBuildLVs()` function.